### DISCUSSION TITLE
Is it possible for partial files to contain custom tags?

### DISCUSSION THREAD

**Discussion by realsama** [markdoc/markdoc#354] [Q&A]
Hello!

I am trying to use Markdoc's partials to write a website but I notice that my custom tags don't get rendered properly. Below is an example.

```
+page.md

{% partial file="base.md" /%}
```

```
base.md

{% component name="users" %}
{% partial file="users.md" /%}
{% /component %}
```

```
users.md

# This is a user

{% code text="Create a user" %}

```python
import os

key = os.environ["SECRET_KEY"]

....

{% /code %}
```

The custom code in the "users.md" won't get transformed but rendered as-is on the generated HTML.

Is what I am trying to do possible? Thanks!

**Comment by rpaul-stripe** [comment]
When you say that the content in `users.md` is being rendered as-is do you mean that the `{% code ...` syntax is showing up in the output? I'm having some trouble reproducing the behavior you are describing. Here's a sample project on Stackblitz with your content, and it appears to be rendering the custom tags as expected: https://stackblitz.com/edit/node-ma7nb9?file=index.mjs

**Comment by realsama** [comment]
@rpaul-stripe  Thank you for your response. The issue is that I am expecting markdoc to render the appropriate tag. I have added my full tags + configuration below.

```
components.svelte

<script lang="ts">
	export let name: string;
	export let page: unknown;
	
	let id: string;

	$: id = name.split('_').slice(-1)[0];

</script>

<div {id}>
	<slot />
</div>
```

```
code.svelte

<script lang="ts">

	export let text: String;

</script>

<div class="terminal space shadow">
	{text}
</div>
```

The expected behaviour is to transform  

```
users.md

# This is a user

{% code text="Create a user" %}

```python
import os

key = os.environ["SECRET_KEY"]

....

{% /code %}
```  

into 

```
<div class="terminal space shadow s-eBgvodDK6F_w">Create a user  <pre data-language="python">import os

key = os.environ["SECRET_KEY"]

  </pre> </div>
```

but what I get is the  below.

```
<component name="users_create"> <h2>How to create a user  </h2> <code text="Create a user"> <pre data-language="python">import os

key = os.environ["SECRET_KEY"]

  </pre> </code> </component>
```


From the above, you can see that markdoc is outputting a <component /> tag that doesn't exist. Hope this gives you more insight into the issue.

**Comment by rpaul-stripe** [comment]
Markdoc doesn't provide a Svelte renderer, so I think the issue you are encountering relates to whatever third-party Markdoc plugin for Svelte you are using. Markdoc itself natively handles tags in partials correctly. We aren't really equipped to answer Svelte-specific problems.
